### PR TITLE
Switch from glide to dep

### PR DESCRIPTION
**Pull request by y0ssar1an**
* flatten dep tree. delete nested `vendor/` directories which cause insidious bugs
* vendor `golint` and `certdump`
* remove prerequisites section from README
  * prereqs are vendored now
* update `.travis.yml` to test on Go 1.8 and Go 1.9

I uncovered a pretty nasty dependency issue while I was doing this. `cfssl_trust` vendors `cfssl` which vendors an old version of `cfssl_trust`. That's bad 😬 . We have multiple versions of `cfssl_trust` in the same repo. Which version of a `cfssl_trust` are we actually executing at any given time?
 
It would be nice if we could break this dependency cycle and make it so that `cfssl` depends on `cfssl_trust`, and not the other way around. This PR is a step in that direction. With the magic of `dep`, all those evil nested `vendor/` directories are gone. `dep` still thinks `github.com/cloudflare/cfssl` is necessary for these packages:
```
github.com/cloudflare/cfssl/helpers
github.com/cloudflare/cfssl/log
github.com/cloudflare/cfssl/signer
github.com/cloudflare/cfssl/ubiquity
```

I apologize for the amount of code churn in this PR. This needs to be tested thoroughly. Then we'll need to switch `cfssl` to `dep` and update its vendored version of `cfssl_trust`.

UPDATE 2018-01-26: 
* revendored using `dep` 0.4.1, which just landed
* added `[prune]` section to `Gopkg.toml` to delete even more stuff from `vendor/`
* removed support for Go 1.6 to avoid transitive dependency on `golang.org/x/net/context`
  * `context` moved into the stdlib in Go 1.7
  * no longer test on Go 1.6 in `.travis.yml`

UPDATE 2018-01-29:
* not sure if this is worth the time after [#845](https://github.com/cloudflare/cfssl/pull/845). feel free to close this

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,15 @@
 sudo: false
 language: go
 go:
-  - tip
+  - master
```
Comment (y0ssar1an) [inline .travis.yml]: `tip` is a legacy reference coming from the days when the Go project used mercurial instead of git

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+  - 1.9
+  - 1.8
   - 1.7
   - 1.6
 before_script:
```
Comment (y0ssar1an) [inline .travis.yml]: With the release of Go 1.10, I'd recommend dropping support for Go 1.8 and below. Then you can remove the ugly `grep -v /vendor/` from `go test ./... | grep -v /vendor/`, etc. in `test.sh`.

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
-  - go get -u github.com/golang/lint/golint
-install:
-  - go get github.com/Masterminds/glide
-  - glide install
+  - cd vendor/github.com/golang/lint/golint
+  - go install
+  - cd -
```
Comment (y0ssar1an) [inline .travis.yml]: We can vendor `golint` by putting it in the `required` section of `Gopkg.toml`. That's one less network request and one less reason for CI to fail.

File: Gopkg.toml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  name = "github.com/mattn/go-sqlite3"
+  version = "1.2.0"
+
+[[constraint]]
+  name = "github.com/prometheus/client_golang"
+  version = "0.8.0"
+
+[[constraint]]
+  name = "gopkg.in/DATA-DOG/go-sqlmock.v1"
+  version = "1.2.0"
```
Comment (y0ssar1an) [inline Gopkg.toml]: These constraints can be relaxed in a future PR.

File: debian/rules [hunk 5]
```diff
@@ -4,7 +4,5 @@ export DH_OPTIONS
 export DH_GOPKG := github.com/cloudflare/cfssl_trust/cmd/cfssl-trust
 
 %:
-	# go get github.com/Masterminds/glide
-	# glide i
```
Comment (y0ssar1an) [inline debian/rules]: I'm not familiar with debian packaging. Are these safe to remove?

Comment (kisom) [reply debian/rules]: Yep, this was an early attempt at debian packaging that maybe shouldn't have been in the repo yet.

**Comment by cbroglie** [comment]
Hi @y0ssar1an, thanks so much for the PR. We definitely want to move towards dep, and I added a ticket to our internal backlog for us to review, test, and get this merged.

**Comment by cbroglie** [comment]
Handled this in #91

**Pull request closed without merging**